### PR TITLE
FISH-6287 Payara6 concurrency 3.0 upgrade

### PR DESCRIPTION
**Pull request by aubi**
## Description
Concurrency 3.0.0 implementation

### Testing Performed
TCK runs clearly, runner is here: https://github.com/payara/jakartaee-10-tck-runners together with docs.

### Testing Environment
Linux, OpenJDK 11

### CODE REVIEW

File: appserver/concurrent/concurrent-connector/src/main/java/org/glassfish/concurrent/config/ManagedExecutorServiceBase.java [hunk 1]
```diff
@@ -158,4 +158,21 @@ public interface ManagedExecutorServiceBase extends ConfigBeanProxy,
      * @param value allowed object is {@link String }
      */
     void setThreadLifetimeSeconds(String value) throws PropertyVetoException;
+
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-connector/src/main/java/org/glassfish/concurrent/config/ManagedExecutorServiceBase.java:161]: Needs Portions copyright

File: appserver/concurrent/concurrent-impl/pom.xml [hunk 2]
```diff
@@ -55,6 +55,13 @@
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/pom.xml:55]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/InvocationContext.java [hunk 3]
```diff
@@ -52,6 +52,8 @@
 import io.opentracing.SpanContext;
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/InvocationContext.java:52]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/TransactionSetupProviderImpl.java [hunk 4]
```diff
@@ -41,14 +41,14 @@
 package org.glassfish.concurrent.runtime;
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/TransactionSetupProviderImpl.java:41]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/BaseConfig.java [hunk 5]
```diff
@@ -98,5 +98,9 @@ public String getContextInfoEnabled() {
         return contextInfoEnabled;
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/BaseConfig.java:98]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/ManagedScheduledExecutorServiceConfig.java [hunk 6]
```diff
@@ -52,6 +52,7 @@ public class ManagedScheduledExecutorServiceConfig extends BaseConfig {
     private int corePoolSize;
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/ManagedScheduledExecutorServiceConfig.java:52]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/ManagedThreadFactoryConfig.java [hunk 7]
```diff
@@ -48,16 +48,22 @@
 public class ManagedThreadFactoryConfig extends BaseConfig {
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/deployer/ManagedThreadFactoryConfig.java:48]: Portions copyright needs updating

File: appserver/concurrent/concurrent-impl/src/test/java/org/glassfish/concurrent/runtime/ConcurrentRuntimeTest.java [hunk 8]
```diff
@@ -40,46 +40,34 @@
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/test/java/org/glassfish/concurrent/runtime/ConcurrentRuntimeTest.java:40]: Portions copyright needs updating

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/ApplicationNode.java [hunk 9]
```diff
@@ -93,8 +93,9 @@ public class ApplicationNode extends AbstractBundleNode<Application> {
     public final static String SCHEMA_ID_16 = "application_6.xsd";
```
Comment (Pandrex247) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/ApplicationNode.java:93]: Portions copyright needs updating

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/DeploymentDescriptorNode.java [hunk 10]
```diff
@@ -526,9 +526,17 @@ protected void setDescriptorInfo(Object target, String methodName, String value)
             } catch (NumberFormatException nfe) {
```
Comment (Pandrex247) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/DeploymentDescriptorNode.java:526]: Portions copyright needs updating

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/xml/TagNames.java [hunk 11]
```diff
@@ -163,6 +163,29 @@ public interface TagNames {
     public static final String JMS_DESTINATION_PROPERTY_NAME = "name";
```
Comment (Pandrex247) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/xml/TagNames.java:163]: Portions copyright needs updating

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/EJBUtils.java [hunk 12]
```diff
@@ -691,10 +691,18 @@ public static RemoteBusinessWrapperBase createRemoteBusinessObject(
             );
```
Comment (Pandrex247) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/EJBUtils.java:691]: Portions copyright needs updating

File: appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/node/EjbBundleNode.java [hunk 13]
```diff
@@ -95,8 +95,9 @@ public class EjbBundleNode extends AbstractBundleNode<EjbBundleDescriptorImpl> {
     public final static String SCHEMA_ID_30 = "ejb-jar_3_0.xsd";
```
Comment (Pandrex247) [inline appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/node/EjbBundleNode.java:95]: Portions copyright needs updating

File: appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/node/EjbNode.java [hunk 14]
```diff
@@ -66,6 +66,9 @@
 import com.sun.enterprise.deployment.node.ResourceRefNode;
```
Comment (Pandrex247) [inline appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/node/EjbNode.java:66]: Portions copyright needs updating

File: appserver/tests/payara-samples/samples/pom.xml [hunk 15]
```diff
@@ -64,7 +64,7 @@
         <module>jaxrs-rolesallowed</module>
         <module>jaxrs-rolesallowed-servlet</module>
         <module>jaxws-security</module>
-        <module>jaxws-tracing</module>
+        <!--module>jaxws-tracing</module-->
```
Comment (Pandrex247) [inline appserver/tests/payara-samples/samples/pom.xml:67]: Are we commenting these out to pass Jenkins?

File: appserver/web/web-glue/src/main/java/org/glassfish/web/deployment/node/WebCommonNode.java [hunk 16]
```diff
@@ -125,6 +125,10 @@ protected WebCommonNode()  {
         registerElementHandler(new XMLElement(TagNames.JMS_DESTINATION), JMSDestinationDefinitionNode.class, "addResourceDescriptor");
```
Comment (Pandrex247) [inline appserver/web/web-glue/src/main/java/org/glassfish/web/deployment/node/WebCommonNode.java:125]: Portions copyright needs updating

File: nucleus/resources/src/main/java/org/glassfish/resourcebase/resources/api/ResourceConstants.java [hunk 17]
```diff
@@ -142,6 +142,7 @@ public interface ResourceConstants {
     public static String JMS_CONNECTION_FACTORY_DEFINITION_JNDINAME_PREFIX = "__jms_connection_factory_definition/";
```
Comment (Pandrex247) [inline nucleus/resources/src/main/java/org/glassfish/resourcebase/resources/api/ResourceConstants.java:142]: Portions copyright needs updating

File: appserver/concurrent/concurrent-connector/src/main/java/org/glassfish/concurrent/config/ManagedExecutorServiceBase.java [hunk 18]
```diff
@@ -158,4 +158,21 @@ public interface ManagedExecutorServiceBase extends ConfigBeanProxy,
...
+    /**
+     * Gets the value of the context property.
+     *
+     * @return possible object is {@link String }
+     */
+    @Attribute(defaultValue = "" + Integer.MAX_VALUE, dataType = String.class)
+    @Min(value = 0)
+    String getContext();
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-connector/src/main/java/org/glassfish/concurrent/config/ManagedExecutorServiceBase.java:169]: This seems to be a name for JNDI, but its default value is 2147483647?

File: appserver/common/container-common/src/main/java/org/glassfish/javaee/services/JndiLookupNotifier.java [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package org.glassfish.javaee.services;
```
Comment (Pandrex247) [inline appserver/common/container-common/src/main/java/org/glassfish/javaee/services/JndiLookupNotifier.java:40]: Does this actually need to be in a `org.glassfish.javaee` package?
Could it a) not be `fish.payara` and b) not be `jakartaee` (or just `ee`)?

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/cdi/ConcurrentCDIExtension.java [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package org.glassfish.concurrent.cdi;
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/cdi/ConcurrentCDIExtension.java:40]: Does this need to be `org.glassfish`? Could it not be `fish.payara`?

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/cdi/ConcurrentCDIExtension.java [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+import org.glassfish.enterprise.concurrent.AsynchronousInterceptor;
+
+/**
+ * CDI Extension for Jakarta Concurrent implementation in Payara.
+ *
+ * @author Petr Aubrecht <aubrecht@asoftware.cz>
+ */
+public class ConcurrentCDIExtension implements Extension {
+
+    static final Logger log = Logger.getLogger(ConcurrentCDIExtension.class.getName());
```
Comment (Pandrex247) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/cdi/ConcurrentCDIExtension.java:66]: Any particular reason it isn't `private`? It doesn't seem to get used anywhere else

File: appserver/common/container-common/src/main/java/org/glassfish/javaee/services/JndiLookupNotifier.java [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Notification, that a CommonResourceProxy object was looked up.
+ *
+ * @author Petr Aubrecht <aubrecht@asoftware.cz>
+ */
+public interface JndiLookupNotifier {
```
[the remark this answers is not in the document]
Comment (aubi) [reply appserver/common/container-common/src/main/java/org/glassfish/javaee/services/JndiLookupNotifier.java:47]: This is used for any object to be informed, that it was resolved by CommonResourceProxy (and it is used there).
Its usage is an ongoing discussion:
https://github.com/jakartaee/concurrency/pull/212      
https://github.com/jakartaee/concurrency/issues/226
If we will have to save context during JNDI lookup, the implementation uses this interface.

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**